Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Profile View/NonprofitContainerViewController.m

```diff
@@ -27,17 +27,25 @@ @implementation NonprofitContainerViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    self.foundInBasketsTableView.delegate = self;
-    self.foundInBasketsTableView.dataSource = self;
-    self.nonprofit = [[User currentUser] nonprofit];
-//    if (self.nonprofit.verificationFiles == nil) { //FIXME: ISVERIFIED
-//        self.nonprofitIsVerifiedCheckmark.hidden = YES;
-//    }
-    self.nonprofitNameLabel.text = self.nonprofit.nonprofitName;
-    self.nonprofitCategoryLabel.text = self.nonprofit.category;
-    self.totalDonationValueLabel.text = [NSString stringWithFormat:@"$%0.2f", self.nonprofit.totalDonationsValue];
-    self.nonprofitWebsiteLabel.text = self.nonprofit.websiteUrlString;
-    self.nonprofitDescriptionLabel.text = self.nonprofit.nonprofitDescription;
+    if ([[User currentUser] nonprofit] != nil) {
+//        self.foundInBasketsTableView.delegate = self;
```

Comment:
Please avoid commented code

Reply:
What's the best thing to do with code in a case where it isn't working yet, but I'll come back to it at a later point?

Reply:
I would just keep it locally in your computer, either in a separate branch or the git stash. I think the main point is just to keep the master branch always nice and clean.

Reply:
Another part of this I think is keeping commits really focused, so you don't write any code that doesn't work yet in the first place, and all the code in the commit is final and ready to go into master.

Reply:
Something that works for me is to split each task into smaller subtasks, thinking about how I'm going to implement the functionality and in which order, and then just do one commit per subtask. For example if I want to add support to create an entity on the app I could plan like this:

- Add an empty view controller and connect it to the main entry point
- Add all the inputs to the view controller
- Create or update the model class if needed
- Create the instance of the model class from the input fields in the view controller
- POST the created instance to the API
- Setup any success actions or navigate away if needed
- Setup error handling
- Setup autolayout
- Polish the UI
- Add animations if needed

That's just an example of course, it can be split in a lot of different ways.

Reply:
Thanks! This is helpful. Sounds like I should work on really breaking down my tasks so I'm only really working on one issue at a time